src/sagemaker/session.py

```diff
@@ -4352,6 +4352,30 @@ def update_endpoint(self, endpoint_name, endpoint_config_name, wait=True):
             self.wait_for_endpoint(endpoint_name)
         return endpoint_name
 
+    def is_inference_component_based_endpoint(self, endpoint_name):
```

Comment:
Why this function need to inside session? Session is used like a util class here.

Reply:
Session contains similar utilities for making SageMaker API calls. We want the code for making the client calls to be in the session since that'll have the proper credentials.